tap_shiphero/sync.py

```diff
@@ -31,17 +31,47 @@ def persist_records(catalog, stream_id, records):
                 singer.write_record(stream_id, record)
                 counter.increment()
 
-def get_bookmark(state, stream_id, default):
-    return (
-        state
-        .get('bookmarks', {})
-        .get(stream_id, default)
-    )
-
-def set_bookmark(state, stream_id, value):
+def get_bookmark(state, stream_id, field, default):
+    """This function will return:
+    - state[bookmarks][stream_id] **Deprecated**
+    - state[bookmarks][stream_id][field]
+    - `default`
+    """
+    the_bookmark = state.get('bookmarks', {}).get(stream_id, default)
+
+    if isinstance(the_bookmark, string):
+        return the_bookmark
+    elif isinstance(the_bookmark, dict):
+        return the_bookmark.get(field, default)
+
+def set_bookmark(state, stream_id, field, value):
+    """Structure of the state dictionary:
+    {
+      "bookmarks": {
+        "stream1" : {
+          "datetime": "2019-01-01T00:00:00Z",
+          "page": 75
+        },
+        "stream2" : {
+          "datetime": "2019-01-01T00:00:00Z",
+          "page": 75
+        },
+        ...
+      }
+      ...
+    }
+    """
     if 'bookmarks' not in state:
         state['bookmarks'] = {}
-    state['bookmarks'][stream_id] = value
+
+    # This is the old style, so convert it to a dict
+    if isinstance(state['bookmarks'][stream_id], string):
+        state['bookmarks'][stream_id] = {}
```

Comment:
We should log info here so we can eventually delete the code once all bookmarks are converted.